sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
I think there are two issues:

1. When required schema is empty. This is why the jenkins test fails. we're working on fix it.

2. If `actualSchema` selects few columns. We noticed that. The column `_corrupted_record` is different when the selected columns are different. But correctly we treat it as a designed behavior. Not sure if this is an issue we need to fix.